common/achievement/evaluate.ts

```diff
@@ -16,10 +16,11 @@ async function _evaluateAchievement(
     userId: string,
     condition: string,
     dataAggregation: ConditionDataAggregations,
-    metrics: string[],
     recordValue: number,
-    relation?: string | null
-): Promise<EvaluationResult> {
+    relation?: string
+): Promise<EvaluationResult | undefined> {
+    // We only care about metrics that are used for the data aggregation
+    const metrics = Object.values(dataAggregation).map((entry) => entry.metric);
```

Comment:
Off topic to this PR: Why is there achievementTemplate.metrics at all then? Also wouldn't it be cleaner to just pass in the achievement template in the parameter?

Reply:
> Why is there achievementTemplate.metrics at all then?

That's a good point. I think the main idea was to have a field, which can be quried, like
```
SELECT * FROM achievement_templates WEHRE metric in (....)
```
Nevertheless, afaics it's not used like this right now.
https://github.com/corona-school/backend/blob/feat/achievements/common/achievement/template.ts#L48

> Also wouldn't it be cleaner to just pass in the achievement template in the parameter?

Why do you think so?

Reply:
Cause its "evaluate achievement", why wouldn't it need the achievement template as a whole?

Reply:
>  I think the main idea was to have a field, which can be queried

I see, I think this is now obsolete as we cache them in memory anyways. Also I fear that this could be a source of bugs if they deviate and some parts of the code look at one or the other

Reply:
> I see, I think this is now obsolete as we cache them in memory anyways. Also I fear that this could be a source of bugs if they deviate and some parts of the code look at one or the other

I move the comment to the main PR

> Cause its "evaluate achievement", why wouldn't it need the achievement template as a whole?

I don't know. I like that you see the important information that you from the achievement right in the function definition. I have at least the feeling that it's easier to understand.
" We want to evaluate the achievement, and therefore we need a condition, the record value and the data aggr."

Reply:
Also reasonable